AdobeMediaGallery/Test/Unit/Model/Keyword/Command/SaveAssetKeywordsTest.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\AdobeMediaGallery\Test\Unit\Model\Keyword\Command;
+
+use Magento\AdobeMediaGallery\Model\Keyword\Command\SaveAssetKeywords;
+use Magento\AdobeMediaGallery\Model\Keyword\Command\SaveAssetLinks;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\DataObject;
+use Magento\Framework\DB\Adapter\Pdo\Mysql;
+use Magento\Framework\DB\Select;
+use Magento\Framework\Exception\CouldNotSaveException;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * SaveAssetKeywordsTest.
+ */
+class SaveAssetKeywordsTest extends TestCase
+{
+    /**
+     * @var SaveAssetKeywords
+     */
+    private $sut;
+
+    /**
+     * @var ResourceConnection|MockObject
+     */
+    private $resourceConnectionMock;
+
+    /**
+     * @var Mysql|MockObject
+     */
+    private $connectionMock;
+
+    /**
+     * @var SaveAssetLinks|MockObject
+     */
+    private $saveAssetLinksMock;
+
+    /**
+     * SetUp
+     */
+    public function setUp(): void
+    {
+        $this->resourceConnectionMock = $this->createMock(ResourceConnection::class);
+        $this->saveAssetLinksMock = $this->createMock(SaveAssetLinks::class);
+        $this->connectionMock = $this->getMockBuilder(Mysql::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->sut = new SaveAssetKeywords(
+            $this->resourceConnectionMock,
+            $this->saveAssetLinksMock
+        );
+    }
+
+    /**
+     * Test saving the asset keywords
+     *
+     * @dataProvider assetKeywordsDataProvider
+     *
+     * @param array $keywords
+     * @param int $assetId
+     * @param array $items
+     */
+    public function testAssetKeywordsSave(array $keywords, int $assetId, array $items): void
+    {
+        $this->prepareResourceConnection();
+        $this->connectionMock->expects($this->once())
+            ->method('insertArray')
+            ->with(
+                'media_gallery_keyword',
+                ['keyword'],
+                $items,
+                2
+            );
+
+        $this->sut->execute($keywords, $assetId);
+    }
+
+    /**
+     * Testing throwing exception handling
+     *
+     * @throws CouldNotSaveException
+     */
+    public function testAssetNotSavingCausedByError(): void
+    {
+        $this->resourceConnectionMock
+            ->method('getConnection')
+            ->willThrowException((new \Exception()));
+        $this->expectException(CouldNotSaveException::class);
+
+        $this->sut->execute([], 1);
+    }
+
+    /**
+     * Preparing the resource connection
+     */
+    private function prepareResourceConnection(): void
+    {
+        $selectStub = $this->createMock(Select::class);
+        $selectStub->method('from')->willReturnSelf();
```

Comment:
All those methods are not called in the class under test. It is necessary to keep only `getConnection` method call expectation and ideally move it to the `setUp` method

Reply:
Actually we have the `getKeywordIds` private method, that fetches the keyword ids. The reason of why I moved configuring `resourceConnection` separately, is that if an Exception is thrown, then we don't need the `connection` at all.